compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt

```diff
@@ -46,29 +46,54 @@ private class DisplayLinkConditions(
         }
 
     /**
-     * Indicates that scene is invalidated and next display link callback will draw
+     * When CADisplayLink invokes a callback, this value is checked to decide if draw needs to be dispatched
      */
-    var needsRedrawOnNextVsync: Boolean = false
+    val needsRedrawOnNextVsync: Boolean
+        get() = scheduledRedrawsCount > 0
+
+    /**
+     * Indicates that application is running foreground now
+     */
+    var isApplicationActive: Boolean = false
         set(value) {
             field = value
 
             update()
         }
 
     /**
-     * Indicates that application is running foreground now
+     * Number of subsequent vsync that will issue a draw
      */
-    var isApplicationActive: Boolean = false
+    private var scheduledRedrawsCount = 0
         set(value) {
             field = value
 
             update()
         }
 
+    /**
+     * Notify that vsync has issued a draw dispatch
+     */
+    fun onDrawStart() {
+        check(scheduledRedrawsCount >= 1)
+        scheduledRedrawsCount -= 1
+    }
+
+    /**
+     * Mark next [FRAMES_COUNT_TO_SCHEDULE_ON_NEED_REDRAW] frames to issue a draw dispatch and unpause displayLink if needed.
+     */
+    fun needRedraw() {
+        scheduledRedrawsCount = FRAMES_COUNT_TO_SCHEDULE_ON_NEED_REDRAW
+    }
+
     private fun update() {
         val isUnpaused = isApplicationActive && (needsToBeProactive || needsRedrawOnNextVsync)
         setPausedCallback(!isUnpaused)
     }
+
+    companion object {
+        const val FRAMES_COUNT_TO_SCHEDULE_ON_NEED_REDRAW = 2
```

Comment:
Need to add comment with explanation of this magic number

Reply:
Done